### ISSUE TITLE
Delay of 6 blocks before ERC20 tokens are converted back to TFT

### ISSUE THREAD

**Issue by LeeSmet** [threefoldfoundation/tfchain#246]
There seems to be no way to subscribe events starting from a certain chain hight. Maybe the contract exposes certain ways to do this, or maybe reload the last 6 blocks on every startup to search for events

**Comment by LeeSmet** [comment]
Seems that the generated contract bindings support a start index in the command, which (should) replay the events even from past blocks.

**Comment by GlenDC** [comment]
> Seems that the generated contract bindings support a start index in the command, which (should) replay the events even from past blocks.

Would be very weird that this was/is not a supported feature, given that in the end the events are just triggers based on the updates of blocks from the chain, I would indeed expect that if you start from a certain height, it would be able to replay those blocks and thus also trigger those events again.

I hope you're right and it is supported indeed, as otherwise we have a big issue.

**Comment by LeeSmet** [comment]
The problem now is that there seems to be no easy way to pick up chain reorg from the (light) client. The events do have a `Removed` bool property in their `raw` structs. While this is usefull for a node which is catching up, it is a property which would imply that it is not updated in real time (for nodes which are already in sync). There also seems to be no `Transaction Receipt` attached unfortunately

**Comment by LeeSmet** [comment]
The `TxHash` of the transaction seems to be encoded in the `Raw` field of the event, so the most straightforward way for now would be to keep track of this hash, then after 6 blocks verify if the hash is still present in the chain on the remote node using the `TransactionByHash` function on the client. There is room to extend these checks if required by checking the block hash and the tx index in the block as well

**Comment by GlenDC** [comment]
`TransactionByHash` is not supported by a light client though, unless the Transaction is in the client db for some reason. I think you're there going to encounter the same issue as I have to finish of the Validation, that we require the LES/v2 GetTxStatus client implementation (as to be able to get the transaction out of the LES/v1 (and already supported) GetBlock functionality).

Unless I'm wrong, in which case my problem is solved as well.

**Comment by LeeSmet** [comment]
Yes you are probably right, it's annoying that the event doesn't include a transaction receipt

**Comment by LeeSmet** [comment]
Even though there is a subscription for `Removed` events, this is not properly implemented in the light client, so manually constructing something like that is also not possible it seems
